presto-main/src/test/java/com/facebook/presto/cost/TestFilterStatsCalculator.java

```diff
@@ -485,6 +486,22 @@ public void testBetweenOperatorFilter()
     }
 
     @Test
+    public void testSymbolEqualsSymbolFilter()
+    {
+        Expression symbolEqualsSymbol = new ComparisonExpression(EQUAL, new SymbolReference("x"), new SymbolReference("x"));
```

Comment:
I wouldn't call `a=a` is a bug. It is a perfectly legal expression. If we can test it on unit level (even as a special case), then why not?